### DISCUSSION TITLE
How to specify an endpoint for a Webhook in settings conf?

### DISCUSSION THREAD

**Discussion by soypunk** [MrThearMan/django-signal-webhooks#14] [General]
The documentation is a little sparse in some spots and I was reviewing the code but I don't fully grok how `SIGNAL_WEBHOOKS` is intended to work. Given the following:

	SIGNAL_WEBHOOKS = {
		"HOOKS": {
			"myapp.models.SampleModel": {
				"CREATE": ...,
				"UPDATE": None,
				"DELETE": None
			},
		},
	}
	
I believe will only fire off a webhook when configured model is signal is invoked and it is a CREATE new record event. I understand the ellipsis is passing in some sorta of default configuration but what or where is that? What I want to do is define the endpoint that should be used with the webhook.

**Comment by MrThearMan** [comment]
Hi, thanks for taking an intrest in the library!

Yes, the documentation is pretty short, but you've understood correctly. Your current setting will fire webhooks for the model on the `post_save` signal, but only when then object is created. The default handler you are wondering about is [signal_webhooks.handlers.default_hook_handler](https://github.com/MrThearMan/django-signal-webhooks/blob/8c70ea728e86bfc1b70fe91fcc7cf844ee1dc70f/signal_webhooks/handlers.py#L111). This is determined in [signal_webhooks.handlers.find_hook_handler](https://github.com/MrThearMan/django-signal-webhooks/blob/8c70ea728e86bfc1b70fe91fcc7cf844ee1dc70f/signal_webhooks/handlers.py#L78C5-L78C22).

However, this is not the place you should setting the endpoint to fire the webhooks to. Instead, you need to create rows in the database for the [signal_webhooks.models.Webhook](https://github.com/MrThearMan/django-signal-webhooks/blob/8c70ea728e86bfc1b70fe91fcc7cf844ee1dc70f/signal_webhooks/models.py#L135)-model. You can do this either in the [admin panel](https://mrthearman.github.io/django-signal-webhooks/setup/#4-add-hooks-in-the-admin-panel) or with the [Django REST Framework-viewset](https://github.com/MrThearMan/django-signal-webhooks/blob/8c70ea728e86bfc1b70fe91fcc7cf844ee1dc70f/signal_webhooks/api/views.py#L11) (undocumented, but should be easy enough to figure out).

Hopefully this helps!

**Reply by soypunk** [reply]
That's helpful, thank you!
